### PR TITLE
helm: remove kube-rbac-proxy and update metrics service

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -49,10 +49,10 @@ config:
 
 envoyGatewayMetricsService:
   ports:
-    - name: https
-      port: 8443
+    - name: http
```
Comment (arkodg) [inline charts/gateway-helm/values.tmpl.yaml:52]: it looks like this was pointing to the controller runtime metrics endpoint earlier
https://book.kubebuilder.io/reference/metrics
https://github.com/envoyproxy/gateway/blob/d65ab597196bba840c9ec87dca2676d6b0502a77/charts/gateway-helm/templates/envoy-gateway-deployment.yaml#L74

are we disabling that now ?

**Review by zirain — DISMISSED** [review]

File: charts/gateway-helm/values.tmpl.yaml (hunk 1 above)
Comment (Xunzhuo) [reply charts/gateway-helm/values.tmpl.yaml:52]: We have collected them together, we have the controller runtime metrics and also our own metrics. They are exposing at 19001 together.

Comment (arkodg) [reply charts/gateway-helm/values.tmpl.yaml:52]: do we then need to delete the above listen address in L75 ?

Comment (Xunzhuo) [reply charts/gateway-helm/values.tmpl.yaml:52]: let me check.

Comment (Xunzhuo) [reply charts/gateway-helm/values.tmpl.yaml:52]: Do not need it anymore.

**Review by shawnh2 — APPROVED** [review]

**Comment by arkodg** [comment]
@Xunzhuo is this needed anymore https://github.com/envoyproxy/gateway/blob/main/charts/gateway-helm/templates/metrics-reader-rbac.yaml ?

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**